### PR TITLE
Project-scoped linting

### PR DESCRIPTION
**Pull request by pepicrft**
### Short description 📝
As you might know, the `fourier lint` commands were scoped to the programming languages: `fourier lint ruby` and `fourier lint swift`. That didn't align with the way we organize the CI pipelines each representing a project. This is the first step towards aligning CI with Fourier's interface and getting rid of the `checks.yml` pipeline. This PR adjusts the `fourier lint` interface to be:

```
  fourier lint all                   # Lint all the code in the repository
  fourier lint backbone              # Lint the Ruby code of the Backbone project
  fourier lint cloud                 # Lint the Ruby code of the Cloud project
  fourier lint cocoapods-interactor  # Lint the Ruby code of the CocoaPods interactor project
  fourier lint fixturegen            # Lint the Swift code of the fixturegen project
  fourier lint fourier               # Lint the Ruby code of the fixturegen project
  fourier lint help [COMMAND]        # Describe subcommands or one specific subcommand
  fourier lint lockfiles             # Ensures SPM and Tuist's generated lockfiles are consistent
  fourier lint tuist                 # Lint the Swift code of the Tuist CLI
  fourier lint tuistbench            # Lint the Swift code of the tuistbench project
  ```
  
  All the commands support passing a `--fix` command, and developers can use `fourier all --fix` to run the lints for all the projects.

### CODE REVIEW

**Comment by pepicrft** [comment]
I still need to add some unit tests to Fourier :)

**Review by fortmarek — APPROVED** [review]
I like the change of thinking (apart from making it easier on CI) this brings - I care about the **project** I am working on, not the **language**.

File: projects/fourier/lib/fourier/services/lint/backbone.rb [hunk 1]
```diff
@@ -0,0 +1,26 @@
+# frozen_string_literal: true
+
+module Fourier
+  module Services
+    module Lint
+      class Backbone < Base
```
Comment (ezraberch) [inline projects/fourier/lib/fourier/services/lint/backbone.rb:6]: There's a lot of code duplication here. Having all of these nearly-identical classes seems unnecessary. Just call the linter with the appropriate directory from `Commands::Lint`.

File: projects/fourier/lib/fourier/utilities/swift_linter.rb [hunk 2]
```diff
@@ -0,0 +1,24 @@
...
+module Fourier
+  module Utilities
+    module SwiftLinter
+      class << self
+        def lint(directories:, fix: false)
+          arguments = [
+            File.join(Fourier::Constants::VENDOR_DIRECTORY, "swiftlint"),
+          ]
+          if fix
+            arguments << "autocorrect" if fix
```
Comment (ezraberch) [inline projects/fourier/lib/fourier/utilities/swift_linter.rb:12]:
```suggestion
          if fix
            arguments << "autocorrect"
```

**Comment by pepicrft** [comment]
Ouch, missed your changes @ezraberch. I'll open a PR adding those

File: projects/fourier/lib/fourier/services/lint/backbone.rb (hunk 1 above)
Comment (pepicrft) [reply projects/fourier/lib/fourier/services/lint/backbone.rb:6]: Part of the duplication has been extracted into `RubocopLinter`. Services were created to have a separation of concerns between commands, which take care of parsing the arguments, and the services, which encapsulate the business logic. If services don't exist we risk coding the logic in the command, which would be the equivalent of placing business logic in views in an MVC pattern.

Comment (ezraberch) [reply projects/fourier/lib/fourier/services/lint/backbone.rb:6]: I agree with the general approach. However, in this case there are 3 entities: the Command, the Service, and the Utility. The real business logic is in the Utility, with the Service just being a pass-though with no real logic. So it seems unnecessary to me.

That said, if you really want the Service to remain, I think keeping the Services as `Lint::Swift` and `Lint::Ruby` is a cleaner approach. Or even a single unified Lint service which lints both (which would avoid missing things in the future).

**Pull request merged**